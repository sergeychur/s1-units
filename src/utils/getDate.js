const weeks = {
	0: 'вс',
	1: 'пн',
	2: 'вт',
	3: 'ср',
	4: 'чт',
	5: 'пт',
	6: 'сб',
};

const months = {
	0: 'января',
	1: 'февраля',
	2: 'марта',
	3: 'апреля',
	4: 'мая',
	5: 'июня',
	6: 'июля',
	7: 'августа',
	8: 'сентября',
	9: 'октября',
	10: 'ноября',
	11: 'декабря',
};

export function getDate(timestamp) {
	const date = new Date(timestamp);
	const day = date.getDate();
	const week = date.getDay();
	const month = date.getMonth();
	const year = date.getFullYear();
	return `${day} ${months[month]}, ${weeks[week]}, ${year} год`;
}
