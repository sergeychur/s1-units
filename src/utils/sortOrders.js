// Array sort doc
// @see https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#Description

export const sortTypes = {
	ITEM_NAMES: 'names',
	DATE: 'date',
	COUNT: 'count',
};

/**
 * Функция сортировки заказов
 * @param {Array} orders - массив заказов
 * @param {string} sortType - выбранный тип сортировки
 */
export const sortOrders = (orders, sortType) => {
	if (!orders || !orders.length || !sortType) {
		return false;
	}

	switch(sortType) {
		case sortTypes.DATE:
			orders.sort(sortByDate);
			break;
		case sortTypes.COUNT:
			orders.sort(sortByItemCount);
			break;
		case sortTypes.ITEM_NAMES:
			orders.sort(sortByItemNames);
			break;
	}
	return true;
};

/**
 * Колбэк для сортировки массива закзаов
 * Сравнивает 2 заказа по именам товаров в заказе
 * @param {Object} order1
 * @param {Object} order2
 */
export const sortByItemNames = (order1, order2) => {
	if (typeof order1 !== 'object' || typeof order2 !== 'object') {
		return 0;
	}

	if (!order1 || !order2) {
		return 0;
	}

	const {items: items1} = order1;
	const {items: items2} = order2;

	if (!items1 || !items2) {
		return 0;
	}

	let result = 0;

	items1.forEach((item, i) => {
			if (item < items2[i]) {
				result = -1;
				return;
			}

			if (item > items2[i]) {
				result = 1;
				return;
			}
	});

	return result;
};

/**
 * Колбэк для сортировки массива заказов
 * Сравнивает 2 заказа по количеству товаров в заказе
 * по возрастанию
 * @param order1
 * @param order2
 */
export const sortByItemCount = (order1, order2) => {
	if (typeof order1 !== 'object' || typeof order2 !== 'object') {
		return 0;
	}

	if (!order1 || !order2) {
		return 0;
	}

	const {items: items1} = order1;
	const {items: items2} = order2;

	if (!items1 || !items2) {
		return 0;
	}

	switch (true) {
		case items1.length < items2.length:
			return -1;
		case items1.length === items2.length:
			return 0;
		default:
			return 1;
	}
};

/**
 * Колбэк для сортировки массива заказов
 * Сравнивает 2 заказа по дате
 * по убыванию (сначала новые заказы)
 * @param order1
 * @param order2
 */
export const sortByDate = (order1, order2) => {
	if (typeof order1 !== 'object' || typeof order2 !== 'object') {
		return 0;
	}

	if (!order1 || !order2) {
		return 0;
	}

	const {date: date1} = order1;
	const {date: date2} = order2;

	if (!date1 || !date2) {
		return 0;
	}

	switch (true) {
		case date1 < date2:
			return 1;
		case date1 > date2:
			return -1;
		default:
			return 0;
	}
};
