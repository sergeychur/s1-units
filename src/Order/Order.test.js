import React from 'react';
import {sortTypes} from '../utils/sortOrders';
import {shallow, configure} from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import Order from './Order.js';
import {fakeOrders} from '../mock/fakeOrders';

configure({ adapter: new Adapter() });

describe('Order component tests', () => {
	it('render order component', () => {
		const order = fakeOrders[0];
		const index = 0;
    	const wrapper = shallow(<Order key={index} order={order}/>);
    	expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('length of items match', () => {
		const order = fakeOrders[0];
		const index = 0;
    	const wrapper = shallow(<Order key={index} order={order}/>);
    	expect(wrapper.find('.Order-item')).toHaveLength(order.items.length)
	});
});