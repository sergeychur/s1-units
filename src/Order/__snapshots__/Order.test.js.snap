// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order component tests render all orders 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Ali Express
    </span>
    <span>
      9 декабря, вс, 2018 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      Утиный пластмасса для показ новый год
    </div>
    <div
      className="Order-item"
      key="1"
    >
      Курица из нержавеющей стали, утка, гусь, голубь, питьевой фонтан
    </div>
    <div
      className="Order-item"
      key="2"
    >
      Новый стиль один розница яйцо для упаковки форма латекс
    </div>
  </div>
</div>
`;

exports[`Order component tests render one order 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Ali Express
    </span>
    <span>
      9 декабря, вс, 2018 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      Утиный пластмасса для показ новый год
    </div>
    <div
      className="Order-item"
      key="1"
    >
      Курица из нержавеющей стали, утка, гусь, голубь, питьевой фонтан
    </div>
    <div
      className="Order-item"
      key="2"
    >
      Новый стиль один розница яйцо для упаковки форма латекс
    </div>
  </div>
</div>
`;

exports[`Order component tests render order component 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Ali Express
    </span>
    <span>
      9 декабря, вс, 2018 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      Утиный пластмасса для показ новый год
    </div>
    <div
      className="Order-item"
      key="1"
    >
      Курица из нержавеющей стали, утка, гусь, голубь, питьевой фонтан
    </div>
    <div
      className="Order-item"
      key="2"
    >
      Новый стиль один розница яйцо для упаковки форма латекс
    </div>
  </div>
</div>
`;

exports[`Order component tests render order component 2`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Ali Express
    </span>
    <span>
      9 декабря, вс, 2018 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      Утиный пластмасса для показ новый год
    </div>
    <div
      className="Order-item"
      key="1"
    >
      Курица из нержавеющей стали, утка, гусь, голубь, питьевой фонтан
    </div>
    <div
      className="Order-item"
      key="2"
    >
      Новый стиль один розница яйцо для упаковки форма латекс
    </div>
  </div>
</div>
`;

exports[`Order component tests shop name matches 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Ali Express
    </span>
    <span>
      9 декабря, вс, 2018 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      Утиный пластмасса для показ новый год
    </div>
    <div
      className="Order-item"
      key="1"
    >
      Курица из нержавеющей стали, утка, гусь, голубь, питьевой фонтан
    </div>
    <div
      className="Order-item"
      key="2"
    >
      Новый стиль один розница яйцо для упаковки форма латекс
    </div>
  </div>
</div>
`;
